### PR TITLE
Remove name bind on instantiation. Upgrade dependency versions

### PR DESCRIPTION
**Pull request by brashidnia-figure**
- Remove `bind_name` usage and instantiation argument (put in empty string in the `ContractInfo` to prevent having to do a full data structure migration)
   - Remove msg to bind the `bind_name` during instantiation
- Upgrade dependencies
- Upgrade rust-optimizer
    - Add Make command for building on M1
    - Update README for the m1 build command

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,6 +4,9 @@ CONTAINER_RUNTIME := $(shell which docker 2>/dev/null || which podman 2>/dev/nul
 .PHONY: all
 all: clean fmt lint test schema optimize
 
+.PHONY: all-arm
+all-arm: clean fmt lint test schema optimize-arm
```
Comment (brashidnia-figure) [inline Makefile:8]: M1 compatibility, worked locally with the new rust optimizer

**Review by leeduan — APPROVED** [review]

**Comment by brashidnia-figure** [comment]
Going to do some more testing on localnet with instantiating/migrating contracts from the current latest version (v0.17.5) to make sure the internet state doesnt break

File: README.md [hunk 2]
```diff
@@ -55,7 +68,7 @@ _note: Address bech32 values may vary._
     
     ```bash
     git clone https://github.com/provenance-io/provenance.git
-    git checkout v1.13.1
+    git checkout v1.14.1
```
Comment (mwoods-figure) [inline README.md:71]: 👍

File: Cargo.toml [hunk 3]
```diff
@@ -28,18 +28,18 @@ overflow-checks = true
...
-rust_decimal = { version = "1.14" }
-schemars = "0.8.1"
-semver = "1.0.3"
-serde = { version = "1.0.127", default-features = false, features = ["derive"] }
-serde_json = "1.0.64"
-thiserror = { version = "1.0.23" }
-uuid = { version= "0.8.2" }
+cosmwasm-std = { version = "1.2.3" }
+cosmwasm-storage = { version = "1.2.3", features = ["iterator"] }
+cw-storage-plus = { version = "1.0.1" }
```
Comment (mwoods-figure) [inline Cargo.toml:33]: It'll be interesting to see what effect this has, if any on gas costs

**Comment by ysong-figure** [comment]
build is failing

File: Cargo.toml (hunk 3 above)
Comment (brashidnia-figure) [reply Cargo.toml:33]: i think our biggest gas cost is actually the fact we use `Bucket` for bid/ask orders which is `cosmwasm-storage`.

We would need to migrate to `HashMap` from `cw-storage-plus` which has a note about how it uses less gas in the docs

**Review by mwoods-figure — APPROVED** [review]

**Comment by brashidnia-figure** [comment]
> build is failing

Build failure is for a set of automated tests to run a local provenance blockchain, build + install the contract, and then run some commands.
seems the result of an external dependency per thread https://figure-group.slack.com/archives/C03P4TQ0W1J/p1679946719192479

Related task to fix the issue https://github.com/provenance-io/provenance-testing-action/issues/9

**Pull request merged**